build.js

```diff
@@ -14,28 +14,32 @@
 
 'use strict';
 
-const css = require('@webref/css');
-const fs = require('fs-extra');
-const idl = require('@webref/idl');
-const path = require('path');
-const prettier = require('prettier');
-const WebIDL2 = require('webidl2');
-const YAML = require('yaml');
+import css from '@webref/css';
+import esMain from 'es-main';
+import fs from 'fs-extra';
+import idl from '@webref/idl';
+import path from 'path';
+import prettier from 'prettier';
+import {fileURLToPath} from 'url';
+import {validate} from 'webidl2';
+import * as YAML from 'yaml';
+
+import customIDL from './custom-idl/index.js';
 
 /* istanbul ignore next */
 const customTests = YAML.parse(
-    fs.readFileSync(
+  await fs.readFile(
     process.env.NODE_ENV === 'test' ?
       './unittest/unit/custom-tests.test.yaml' :
       './custom-tests.yaml',
     'utf8'
-    )
+  )
 );
-const customCSS = require('./custom-css.json');
-const customIDL = require('./custom-idl');
-const customJS = require('./custom-js.json');
 
-const generatedDir = path.join(__dirname, 'generated');
+const customCSS = JSON.parse(await fs.readFile('./custom-css.json'));
+const customJS = JSON.parse(await fs.readFile('./custom-js.json'));
+
+const generatedDir = fileURLToPath(new URL('./generated', import.meta.url));
```

Comment:
Is this really the most straightforward way to get a path relative to the current file with ESM? Going via URLs seems roundabout to me...

Reply:
This is what the BCD conversion does for replacing `__dirname` as well.  I've tried other methods that don't use URLs but I've had no luck.

Reply:
https://stackoverflow.com/questions/46745014/alternative-for-dirname-in-node-when-using-the-experimental-modules-flag points to this solution. I don't think it's very pretty, but alright then :)

Reply:
Actually, elsewhere with `fs.readFile` and `fs.readJson` we already assume that relative paths will work. So let's just assume the same here to avoid the extra stuff:

```suggestion
const generatedDir ='./generated';
```